File: app/src/main/java/com/example/msimple/network/DataAgent.java
package com.example.msimple.network;

import com.example.msimple.Delegate.CategoryDelegate;
import com.example.msimple.Delegate.CurrentProgramDelegate;
import com.example.msimple.Delegate.LoginDelegate;
import com.example.msimple.Delegate.TopicDelegate;

public interface DataAgent {

    void loadCategory(String access_token, String page, CategoryDelegate categoryDelegate);
    void loadTopic(String access_token, int page, TopicDelegate topicDelegate);
    void loadCurrentProgram(String access_token, int page, CurrentProgramDelegate currentProgramDelegate);
    void login(String phone,String password,LoginDelegate loginDelegate);
    void register();


}

File: app/src/main/java/com/example/msimple/network/DataAgent.kt
package com.example.msimple.network

import com.example.msimple.delegate.CategoryDelegate
import com.example.msimple.delegate.CurrentProgramDelegate
import com.example.msimple.delegate.LoginDelegate
import com.example.msimple.delegate.TopicDelegate

interface DataAgent {

    fun loadCategory(access_token: String, page: String, categoryDelegate: CategoryDelegate)
    fun loadTopic(access_token: String, page: Int, topicDelegate: TopicDelegate)
    fun loadCurrentProgram(access_token: String, page: Int, currentProgramDelegate: CurrentProgramDelegate)
    fun login(phone: String, password: String, loginDelegate: LoginDelegate)
    fun register()


}
